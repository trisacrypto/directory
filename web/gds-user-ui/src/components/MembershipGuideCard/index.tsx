import React from 'react';
import { Box, Text, Stack, Button, HStack, VStack } from '@chakra-ui/react';

type MembershipGuideCardProps = {
  stepNumber: number;
  header: string;
  description: string;
  buttonText: string;
};

const MembershipGuideCard = ({
  stepNumber,
  header,
  description,
  buttonText
}: MembershipGuideCardProps) => (
  <Box textAlign="center" width="100%" maxWidth={300} minHeight="100%">
    <Stack gap={'1rem'} backgroundColor="#E5EDF1" p="1rem" height="100%">
      <VStack>
        <Text textAlign="center" fontSize="xl" data-testid="step" textTransform="capitalize">
          Step {stepNumber}
        </Text>
        <Text fontWeight="bold" data-testid="header" textTransform="capitalize">
          {header}
        </Text>
        <Text data-testid="description">{description}</Text>
      </VStack>
      <Box marginTop="auto !important">
        <Button
          variant="solid"
          size="md"
          display="inline-block"
          border="1px solid #221F1F"
          borderRadius={0}
          textTransform="capitalize">
          {buttonText}
        </Button>
      </Box>
    </Stack>
  </Box>
);

export default MembershipGuideCard;
