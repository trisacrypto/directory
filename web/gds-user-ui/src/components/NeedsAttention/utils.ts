export const enum AttentionAction {
  'VERIFY_EMAILS' = 'VERIFY_EMAILS',
  'VERIFY_PHONES' = 'VERIFY_PHONES',
  'VERIFY_ADDRESSES' = 'VERIFY_ADDRESSES',
  'VERIFY_IDENTITIES' = 'VERIFY_IDENTITIES',
  'VERIFY_DOCUMENTS' = 'VERIFY_DOCUMENTS',
  'VERIFY_ALL' = 'VERIFY_ALL',
  'VERIFY_NONE' = 'VERIFY_NONE'
}
