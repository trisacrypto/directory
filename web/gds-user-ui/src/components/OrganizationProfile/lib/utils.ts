export const StatusCheck = (status: string) => {
  switch (status) {
    case 'NO_VERIFICATION':
      return 'Not Verified';
    case 'PENDING_REVIEW':
      return 'Pending Review';
    case 'VERIFIED':
      return 'Verified';
    case 'REJECTED':
      return 'Rejected';
    default:
      return 'Pending Registration';
  }
};
