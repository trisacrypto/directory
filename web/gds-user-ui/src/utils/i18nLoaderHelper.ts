import { i18n } from '@lingui/core';
import { en, fr, de, zh, ja, ru } from 'make-plural/plurals';
import { Locales } from 'types/type';

export const DEFAULT_LOCALE: Locales = 'en';

export async function dynamicActivate(locale: string) {
  const { messages } = await import(`../locales/${locale}/messages`);

  i18n.load(locale, messages);
  i18n.loadLocaleData({
    en: { plurals: en },
    de: { plurals: de },
    fr: { plurals: fr },
    zh: { plurals: zh },
    ja: { plurals: ja }
  });
  i18n.activate(locale);
}
