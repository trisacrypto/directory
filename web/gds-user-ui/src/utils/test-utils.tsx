import { ChakraProvider } from '@chakra-ui/react';
import { render as rtlRender, RenderOptions as RTLRenderOptions } from '@testing-library/react';
import React from 'react';
import { FormProvider, useForm } from 'react-hook-form';
import { Provider } from 'react-redux';
import customTheme from './theme';
import rootReducer from 'application/store/rootReducer';
import { configureStore, Store } from '@reduxjs/toolkit';

type RenderOptions = {
  rhfDefaultValues?: { [x: string]: any };
  store?: Store;
  preloadedState?: StoreState;
  renderOptions?: Omit<RTLRenderOptions, 'wrapper'>;
};

type StoreState = {
  [x: string]: any;
};

export function render(
  ui: React.ReactElement<any, string | React.JSXElementConstructor<any>>,
  {
    rhfDefaultValues,
    preloadedState,
    store = configureStore({ reducer: rootReducer, preloadedState }),
    ...renderOptions
  }: RenderOptions = {}
) {
  function Wrapper({ children }: { children: React.ReactNode }) {
    const methods = useForm({ defaultValues: rhfDefaultValues });
    return (
      <Provider store={store}>
        <ChakraProvider theme={customTheme}>
          <FormProvider {...methods}>{children}</FormProvider>
        </ChakraProvider>
      </Provider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react';
