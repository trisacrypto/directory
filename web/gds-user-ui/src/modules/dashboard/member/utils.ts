import { DirectoryType, VaspDirectoryEnum, DirectoryTypeEnum } from './memberType';
import { convertToCVS, downloadCSV } from 'utils/utils';
import { t } from '@lingui/macro';

export const memberTableHeader = [
  {
    key: 'name',
    label: t`Member Name`
  },
  {
    key: 'first_listed',
    label: t`Joined`
  },
  {
    key: 'last_updated',
    label: t`Last Updated`
  },
  {
    key: 'registered_directory',
    label: t`Network`
  },
  {
    key: 'status',
    label: t`Status`
  }
];
export const getVaspDirectory = (dir: DirectoryType) => {
  return dir === DirectoryTypeEnum.TESTNET ? VaspDirectoryEnum.TESTNET : VaspDirectoryEnum.MAINNET;
};

export const getVaspNetwork = (dir: any) => {
  switch (dir) {
    case (VaspDirectoryEnum.TESTNET, VaspDirectoryEnum.TESTNET_DEV):
      return 'TestNet';
    case (VaspDirectoryEnum.MAINNET, VaspDirectoryEnum.MAINNET_DEV):
      return 'MainNet';
    default:
      return 'MainNet';
  }
};

export const getVaspStatus = (status: number) => {
  switch (status) {
    case 1:
      return t`NO VERIFICATION`;
    case 2:
      return t`SUBMITTED`;
    case 3:
      return t`EMAIL VERIFIED`;
    case 4:
      return t`PENDING REVIEW`;
    case 5:
      return t`REVIEWED`;
    case 6:
      return t`VERIFIED`;
    case 7:
      return t`REJECTED`;
    case 8:
      return t`APPEALED`;
    case 9:
      return t`ERRORED`;
    default:
      return t`NO VERIFICATION`;
  }
};

export const downloadMembers2CVS = (member: any) => {
  const m = member.map((item: any) => {
    return {
      ...item,
      status: getVaspStatus(item.status),
      registered_directory: getVaspNetwork(item.registered_directory)
    };
  });
  const memberCsv = convertToCVS(m, memberTableHeader as ITableHeader[]);
  downloadCSV(memberCsv, 'members');
};

