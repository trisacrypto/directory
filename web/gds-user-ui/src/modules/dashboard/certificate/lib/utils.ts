import { basicDetailsValidationSchema } from 'modules/dashboard/certificate/lib/basicDetailsValidationSchema';
import { legalPersonValidationSchemam } from 'modules/dashboard/certificate/lib/legalPersonValidationSchema';
import { contactsValidationSchema } from 'modules/dashboard/certificate/lib/contactsValidationSchema';
import { trisaImplementationValidationSchema } from 'modules/dashboard/certificate/lib/trisaImplementationValidationSchema';
import { trixoQuestionnaireValidationSchema } from 'modules/dashboard/certificate/lib/trixoQuestionnaireValidationSchema';

export const isBasicDetailsCompleted = async (data: any) => {
  try {
    const r = await basicDetailsValidationSchema.validate(data);
    console.log(r);
    return true;
  } catch (err) {
    return false;
  }
};

export const isLegalPersonCompleted = async (data: any) => {
  try {
    const r = await legalPersonValidationSchemam.validate(data);
    console.log(r);
    return true;
  } catch (err) {
    console.log('legal person validation failed', err);
    return false;
  }
};

export const isContactsCompleted = async (data: any) => {
  try {
    const r = await contactsValidationSchema.validate(data);
    console.log(r);
    return true;
  } catch (err) {
    console.log('contacts validation failed', err);
    return false;
  }
};

export const isTrisaImplementationCompleted = async (data: any) => {
  try {
    await trisaImplementationValidationSchema.validate(data);
    return true;
  } catch (err) {
    return false;
  }
};

export const isTrixoQuestionnaireCompleted = async (data: any) => {
  try {
    await trixoQuestionnaireValidationSchema.validate(data);
    return true;
  } catch (err) {
    return false;
  }
};
