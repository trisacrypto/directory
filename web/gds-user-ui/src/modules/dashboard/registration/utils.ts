import { getDefaultValue } from 'components/BasicDetailsForm/validation';
import { getRegistrationDefaultValues } from 'modules/dashboard/certificate/lib';
import {
  postRegistrationData,
  getRegistrationData,
  getSubmissionStatus
} from 'modules/dashboard/registration/service';
import { handleError } from 'utils/utils';

export const postRegistrationValue = (data: any) => {
  console.log('[postRegistrationValue]', data);
  return new Promise((resolve, reject) => {
    postRegistrationData(data)
      .then((res) => {
        console.log('[postRegistrationData]', res);
        if (res.status === 204) {
          resolve(res);
        } else {
          reject(res);
        }
      })
      .catch((err) => {
        console.log('[postRegistrationData]', err);
        reject(err);
      });
  });
};
export const getRegistrationDefaultValue = async () => {
  try {
    const regData = await getRegistrationData();
    if (regData.status === 200 && Object.keys(regData.data).length > 0) {
      return regData.data;
    }
    const defaultValue: any = localStorage.getItem('certificateForm');
    if (defaultValue) {
      const val = JSON.parse(defaultValue);
      const postData = await postRegistrationData(val);
      if (postData.status === 204) {
        const getData = await getRegistrationData();
        localStorage.removeItem('certificateForm');
        return getData.data;
      }
    }
    const v = getRegistrationDefaultValues();
    await postRegistrationValue(v);
    return getRegistrationDefaultValues();
  } catch (err: any) {
    handleError(err, 'failed to get registration data');
  }
};

export const setRegistrationDefaultValue = () => {
  console.log('[setRegistrationDefaultValue]');
  const defaultValue: any = getRegistrationDefaultValues();
  return new Promise((resolve, reject) => {
    postRegistrationData(defaultValue)
      .then((res) => {
        // console.log('[default postRegistration value]', res);
        if (res.status === 204) {
          resolve(res);
        } else {
          reject(res);
        }
      })
      .catch((err) => {
        handleError(err, 'failed to post registration value');
        reject(err);
      });
  });
};

// get registration data from backend and download the file
export const downloadRegistrationData = async () => {
  try {
    const regData = await getRegistrationData();
    if (regData.status === 200 && Object.keys(regData.data).length > 0) {
      const blob = new Blob([JSON.stringify(regData.data)], { type: 'application/json' });
      const url = window.URL.createObjectURL(blob);
      const link = document.createElement('a');
      link.href = url;
      link.setAttribute('download', 'registration.json');
      document.body.appendChild(link);
      link.click();
    }
  } catch (err: any) {
    handleError(err, 'failed to get registration data');
  }
};

// load default stepper

export const getDefaultStepper = async () => {
  try {
    const [regData, regStatus] = await Promise.all([getRegistrationData(), getSubmissionStatus()]);
    console.log('[regStatus]', regStatus.data);
    if (regData.status === 200 && Object.keys(regData.data).length > 0) {
      return {
        currentStep: regData.data.state.current,
        steps: regData.data.state.steps,
        lastStep: null,
        hasReachSubmitStep: regData.data.state.ready_to_submit,
        testnetSubmitted: regStatus?.data?.testnetSubmitted || false,
        mainnetSubmitted: regStatus?.data?.mainnetSubmitted || false
      };
    }
    const defaultValue: any = {
      current: 1,
      steps: [
        {
          key: 1,
          status: 'progress'
        }
      ]
    };
    // update registrations state object
    const postData = await postRegistrationData({ state: defaultValue });
    if (postData.status === 204) {
      const getData = await getRegistrationData();
      return {
        currentStep: getData.data.state.current,
        steps: getData.data.state.steps,
        lastStep: null,
        hasReachSubmitStep: false,
        testnetSubmitted: false,
        mainnetSubmitted: false
      };
    }
  } catch (err: any) {
    handleError(err, 'failed to get stepper data');
  }
};

// // load default stepper without async call
// export const loadDefaultStepperSync = () => {
//   return Promise.resolve({
//     currentStep: 1,
//     steps: [
//       {
//         key: 1,
//         status: 'progress'
//       }
//     ],
//     lastStep: null,
//     hasReachSubmitStep: false
//   });
// };

// set stepper data
export const getRegistrationAndStepperData = async () => {
  try {
    const [regData, regStatus] = await Promise.all([
      getRegistrationDefaultValue(),
      getSubmissionStatus()
    ]);
    if (regData) {
      const response: any = {
        registrationData: regData,
        stepperData: {
          currentStep: regData?.state?.current || 1,
          steps: regData?.state?.steps || [{ key: 1, status: 'progress' }],
          lastStep: null,
          hasReachSubmitStep: regData?.state?.ready_to_submit || false,
          testnetSubmitted: !!regStatus?.data?.testnet_submitted,
          mainnetSubmitted: !!regStatus?.data?.mainnet_submitted
        }
      };
      return response;
    }
  } catch (err: any) {
    handleError(err, 'failed to get stepper data');
  }
};
