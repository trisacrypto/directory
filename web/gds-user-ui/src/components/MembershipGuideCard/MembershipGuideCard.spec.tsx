import { dynamicActivate } from 'utils/i18nLoaderHelper';
import { act, render, screen } from 'utils/test-utils';
import MembershipGuideCard from '.';

describe('<MembershipGuideCard />', () => {
  beforeAll(() => {
    act(() => {
      dynamicActivate('en');
    });
  });

  it('should render correctly props', () => {
    const { container } = render(
      <MembershipGuideCard
        stepNumber={2}
        header="Create Your Account"
        description="Create your TRISA account with your VASP email address"
        buttonText="Learn More"
      />
    );

    expect(screen.getByTestId('step').textContent).toBe('Step 2');
    expect(screen.getByTestId('description').textContent).toBe(
      'Create your TRISA account with your VASP email address'
    );
    expect(screen.getByTestId('header').textContent).toBe('Create Your Account');
    expect(screen.getByRole('button', { name: 'Learn More' })).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(
      <MembershipGuideCard
        stepNumber={2}
        header="Create Your Account"
        description="Create your TRISA account with your VASP email address"
        buttonText="Learn More"
      />
    );
    expect(container).toMatchSnapshot();
  });
});
