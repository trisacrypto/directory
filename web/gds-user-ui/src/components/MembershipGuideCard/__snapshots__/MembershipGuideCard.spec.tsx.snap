// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MembershipGuideCard /> should match snapshot 1`] = `
<div>
  <div
    class="css-wsis5l"
  >
    <div
      class="chakra-stack css-ly7s1u"
    >
      <div
        class="chakra-stack css-owjkmg"
      >
        <p
          class="chakra-text css-wz7pgp"
          data-testid="step"
        >
          Step 
          2
        </p>
        <p
          class="chakra-text css-8cgwqi"
          data-testid="header"
        >
          Create Your Account
        </p>
        <p
          class="chakra-text css-0"
          data-testid="description"
        >
          Create your TRISA account with your VASP email address
        </p>
      </div>
      <div
        class="css-14d2w9n"
      >
        <button
          class="chakra-button css-vx0ruo"
          type="button"
        >
          Learn More
        </button>
      </div>
    </div>
  </div>
</div>
`;
